src/Api.Rest.Dtos/Data/CatalogDto.cs

```diff
@@ -61,19 +64,10 @@ public ushort[] ValidAttributes
 		/// Gets or sets the list of catalog entries that belong to this catalog.
 		/// </summary>
 		[JsonProperty( "catalogEntries" )]
-		public CatalogEntryDto[] CatalogEntries
+		public IReadOnlyList<CatalogEntryDto> CatalogEntries
 		{
 			[NotNull] get => _CatalogEntries;
-			set
-			{
-				_Dictionary.Clear();
-
-				_CatalogEntries = value ?? Array.Empty<CatalogEntryDto>();
-				foreach( var entry in _CatalogEntries )
-				{
-					_Dictionary[ entry.Key ] = entry;
-				}
-			}
+			set => _CatalogEntries = value ?? Array.Empty<CatalogEntryDto>();

```

Comment:
I see the possibility that after setting the entries to a new list the _Dictionary still contains the old values, since it was not cleared / set to null to force a refresh on the next access.

Reply:
You are right. Thank you.

Reply:
Fixed.